### PR TITLE
Fix assertion failure in set_subqueryscan_references

### PR DESCRIPTION
**Pull request by xiong-gang**
For rollup/cube grouping query, planner adds an append node on
the top of rollups, we need to have root->simple_rel_array includes
all the rollups, also set the subroot and subplan for each rollups
if there's an SubqueryScan node in the rollup.

### CODE REVIEW

File: src/backend/optimizer/plan/plangroupext.c [hunk 1]
```diff
@@ -1858,13 +1911,34 @@ generate_list_subplans(PlannerInfo *root, int num_subplans,
...
+	{
+		int arraySize = root->simple_rel_array_size;
+		RelOptInfo **relArray = (RelOptInfo **) palloc0(sizeof(RelOptInfo *) * arraySize);
+		for (j = 0; j < arraySize; j++)
+		{
+			RelOptInfo *rel = NULL;
+			if (root->simple_rel_array[j] != NULL)
+			{
+				rel = makeNode(RelOptInfo);
+				memcpy(rel, root->simple_rel_array[j], sizeof(RelOptInfo));
```
Comment (guofengrichard) [inline src/backend/optimizer/plan/plangroupext.c:1934]: Is `memcpy` enough here?  Should we consider to use `copyObject` instead?

File: src/backend/optimizer/plan/plangroupext.c [hunk 2]
```diff
@@ -134,6 +135,12 @@ typedef struct GroupExtContext
 	List *pathkeys;
 } GroupExtContext;
 
+typedef struct SubqueryScanWalkerContext
+{
+	SubqueryScan *firstSubqueryScan;
+} SubqueryScanWalkerContext;
+
```
Comment (guofengrichard) [inline src/backend/optimizer/plan/plangroupext.c:142]: Please add a GPDB tag to indicate this is added by GPDB.

File: src/backend/optimizer/plan/plangroupext.c [hunk 3]
```diff
@@ -2162,16 +2236,98 @@ append_colIdx(AttrNumber *colIdx, Oid *operators, int numcols, int colno,
...
+		root->simple_rte_array[i] = lfirst(l);
+		rel = build_simple_rel(root, i, RELOPT_BASEREL);
+		rel->subroot = NULL;
+		rel->subplan = NULL; 
+
+
+		if (i - 1  >= list_length(subplans))
+			continue;
+		splan = (SubqueryScan *)list_nth(subplans, i - 1);
+		sroot = (PlannerInfo *)list_nth(subroots, i - 1);
```
Comment (paul-guo-) [inline src/backend/optimizer/plan/plangroupext.c:2313]: We should use forthree() for rtable, subplans, subroots.

File: src/backend/optimizer/plan/plangroupext.c (hunk 1 above)
Comment (xiong-gang) [reply src/backend/optimizer/plan/plangroupext.c:1934]: we can't do copyObject on RelOptInfo. I think memcpy should be fine here.

File: src/backend/optimizer/plan/plangroupext.c (hunk 2 above)
Comment (xiong-gang) [reply src/backend/optimizer/plan/plangroupext.c:142]: Not necessarily. This file is GPDB only

**Pull request merged**